dockerfiles/Dockerfile.tensorrt

```diff
@@ -25,6 +25,6 @@ RUN git clone --single-branch --branch ${ONNXRUNTIME_BRANCH} --recursive ${ONNXR
     cp onnxruntime/dockerfiles/LICENSE-IMAGE.txt /code/LICENSE-IMAGE.txt &&\
     cp onnxruntime/ThirdPartyNotices.txt /code/ThirdPartyNotices.txt &&\
     cd onnxruntime &&\
-    /bin/sh ./build.sh --parallel --cuda_home /usr/local/cuda --cudnn_home /usr/lib/x86_64-linux-gnu/ --use_tensorrt --tensorrt_home /workspace/tensorrt --config Release --build_wheel --update --build --cmake_extra_defines ONNXRUNTIME_VERSION=$(cat ./VERSION_NUMBER) &&\
+    /bin/sh ./build.sh --parallel --build_shared_lib --cuda_home /usr/local/cuda --cudnn_home /usr/lib/x86_64-linux-gnu/ --use_tensorrt --tensorrt_home /workspace/tensorrt --config Release --build_wheel --update --build --cmake_extra_defines ONNXRUNTIME_VERSION=$(cat ./VERSION_NUMBER) &&\
```

Comment:
Why this change is needed?

Reply:
For EP Perf Testing, we run in a docker container built from this dockerfile. The [memory leak testing](https://github.com/microsoft/onnxruntime/tree/master/onnxruntime/python/tools/tensorrt/perf/mem_test) requires the onnxruntime shared library to exist. However, this is a work in progress and we may only run the memory leak test with a different dockerfile. I was previously trying to combine two separate build steps.